lib/sensu-plugin/utils.rb

```diff
@@ -46,6 +46,31 @@ def net_http_req_class(method)
         end
       end
 
+      # Use API query parameters to paginate HTTP GET requests,
+      # iterating over the results until an empty set is returned.
+      #
+      # @param path [String]
+      # @param options [Hash]
+      # @return [Array]
+
+      def paginated_get(path, options = {})
+        limit = options.fetch('limit', 500)
+        offset = 0
+        results = []
+        loop do
```

Comment:
is there any concept of rate limiting? Even with pagination you might still want/need to rate limit and I am just trying to think a bit ahead as this could make a lot of requests very quickly. On a different note should we consider putting in a timeout?

Reply:
I think concerns around rate limiting and timeout are reasonable, but they will require changes in existing methods (e.g. `api_request`), so I'd rather they were implemented separately.

Reply:
Sounds good to me, just wanted to pick your brain on it.